crates/cli/src/package.rs

```diff
@@ -77,9 +78,52 @@ async fn internal_fetch(
     Ok(())
 }
 
+pub async fn install_single_package_to_virtual_store(
+    tarball_cache: &Cache,
+    http_client: &Client,
+    config: &'static Npmrc,
+    dependency_path: &DependencyPath,
+    package_snapshot: &PackageSnapshot,
+) -> Result<(), PackageManagerError> {
+    let PackageSnapshot { resolution, .. } = package_snapshot;
+    let LockfileResolution::Registry(registry_resolution) = resolution else {
+        panic!("Only TarballResolution is supported at the moment, but {dependency_path} requires {resolution:?}");
+    };
+
+    let DependencyPath { custom_registry, package_specifier } = dependency_path;
+    let registry = custom_registry.as_ref().unwrap_or(&config.registry);
+    let registry = registry.strip_suffix('/').unwrap_or(registry);
+    let PkgNameVerPeer { name, suffix: ver_peer } = package_specifier;
+    let version = ver_peer.version();
+    let bare_name = name.bare.as_str();
+    let tarball_url = format!("{registry}/{name}/-/{bare_name}-{version}.tgz");
```

Comment:
The tarball_url does not always have a standard format. In that case it is stored in `resolution.tarball`. So if there is a `resolution.tarball` field, it should be used for `tarball_url`.

Reply:
If there is a `tarball` field, it would panic:

https://github.com/pnpm/pacquet/blob/b73d1c9f518413b5655002440d8bfdda26995779/crates/cli/src/package.rs#L89-L91

I don't intend on implementing custom registries or custom tarballs in this PR.

Reply:
Anyway, since implementing this is easy, I did just that: https://github.com/pnpm/pacquet/pull/128/commits/80aa1960ce9419972e884b510fb47811fd2cbfed